### PR TITLE
Benchmark script

### PR DESCRIPTION
**Pull request by lafont-e**
Uses vegeta to make request to a 0-metadata server
	When run, creates 4 files in the same directory with graph data

### CODE REVIEW

File: test/bench.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+echo -n "" > ${file}
+for i in $(seq 1 1000); do 
+    body="{\"uid\": $i,\"addr\": \"home\", \"alias\": [\"jsmith\"], \"keyPub\": [\"123\"]}"
+    echo $body > "./targets/jsbody.${i}"
+    echo "POST ${url}/user/${i}" >> ${file}
+    echo "Content-Type: application/json" >> ${file}
+    echo "@./targets/jsbody.${i}" >> ${file}
+    echo >> ${file}
+done 
+
```
Comment (zaibon) [inline test/bench.sh]: missing `echo POST` here ?

**Comment by zaibon** [comment]
Seems also one of the test on travis is failing

**Comment by zaibon** [comment]
And can we also print the test report on the terminal after the benchmark is done.
So we get plot and simple text we can read directly too

File: test/bench.sh (hunk 1 above)
Comment (lafont-e) [reply test/bench.sh]: Yes, but I was thinking that this echo maybe is not needed, what do you prefer, to add the missing echo or delete the other ones

File: test/bench.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+echo "LIST"
+echo "GET ${url}/user" | \
+vegeta attack  -duration=$secs | \
+vegeta report -reporter=plot > plot.LIST.html
+
+echo "DELETE"
+file="./targets/DELETE"
+echo -n "" > ${file}
+for i in $(seq 1 1000); do 
+    echo "DELETE ${url}/user/${i}"
```
Comment (zaibon) [inline test/bench.sh]: missing the redirection to write into the file here ?

File: test/bench.sh (hunk 1 above)
Comment (zaibon) [reply test/bench.sh]: add the missing one, I like to see progress

**Comment by lafont-e** [comment]
About the Makefile

Before pushing the bench.sh I already tried to add a bench: rule in the makefile, but it will not work as I need to change directory and I need to pass a parameter to the bench.sh script, none of those features are simple in makefiles

File: test/bench.sh [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+[ "$1" = "" ] && usage
+
+url="$1"
+
+secs="10s"
+targets="./targets"
+maxrecords=999
+
+
+echo -n "POST..."
```
Comment (zaibon) [inline test/bench.sh]: Any reason this is not on line 40?

File: test/bench.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+
+a=$(which vegeta) 
+[ $? -ne 0 ] && echo "Vegeta Dependency missing" && exit 1
+
+[ "$1" = "" ] && usage
+
+url="$1"
+
+secs="10s"
```
Comment (zaibon) [inline test/bench.sh]: Maybe we could make this configurable like the URL.
Also would be nice to also control the rate of vegeta

**Comment by zaibon** [comment]
Why exactly do you need to change directory? Can't you just call the script and with relative path?
And for parameter, we can solve it by having default values for all arguments. So it has a working default behavior

File: test/bench.sh (hunk 3 above)
Comment (lafont-e) [reply test/bench.sh]: no, it was there initially and as the "for"  is also preparation for POST seems natural, but changed it to his proper line

File: test/bench.sh (hunk 4 above)
Comment (lafont-e) [reply test/bench.sh]: The default value for the URL not sure is going to work as it is the URL for the 0-metadata server, and not sure how much people is going to install it on the same URL, anyways will make a default value for it, we can always change it if needed.
For the script with relative path, it's ok, I just need that the targets directory falls in a sure place

File: Makefile [hunk 5]
```diff
@@ -1,4 +1,7 @@
 
+bench:
+	/bin/bash test/bench.sh http://172.17.0.2:8888
```
Comment (zaibon) [inline Makefile]: use localhost as default value and the same port as the default port of the server

**Comment by lafont-e** [comment]
Changed the default URL and small cosmetic changes to the bench script

**Pull request merged**